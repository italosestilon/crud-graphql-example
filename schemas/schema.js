import {GraphQLSchema} from 'graphql';

import ArtistType from '../types/artist';
import Artist from '../models/artist';

import QueryType from '../queries/index';
import MutationType from '../mutations/index';

export const schema = new GraphQLSchema({
	query: QueryType,
	mutation: MutationType
});