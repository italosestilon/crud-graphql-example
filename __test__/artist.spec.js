import Artist from "../models/artist";
import { graphql } from "graphql";
import schema from "../schemas/schema";

import {
  connect,
  clearDatabase,
  disconnect,
  sanitizeObject
} from "../test/helper";

beforeAll(connect);
afterEach(clearDatabase);
afterAll(disconnect);

it("should not save an empty artist", async () => {
  try {
    const artist = new Artist({});
    await artist.save();
  } catch (err) {
    expect(err).toMatchSnapshot();
  }
});

it("should save new artist", async () => {
  try {
    const artist = new Artist({
      name: "Lana Del Rey"
    });
    await artist.save();
    expect(sanitizeObject(artist._doc)).toMatchSnapshot();
  } catch (err) {
    expect(err).toMatchSnapshot();
  }
});

it("should retrieve an artist", async () => {
  try {
    const artist = new Artist({
      name: "Lana Del Rey"
    });

    await artist.save();

    const retrievedArtist = await Artist.findById(artist.id);

    expect(sanitizeObject(retrievedArtist._doc)).toMatchSnapshot();
  } catch (err) {
    expect(err).toMatchSnapshot();
  }
});

it("should delete an artist", async () => {
  try {
    const artist = new Artist({
      name: "Lana Del Rey"
    });

    await artist.save();
    const retrievedArtist = await Artist.findByIdAndDelete(artist.id);
    expect(sanitizeObject(retrievedArtist._doc)).toMatchSnapshot();
  } catch (err) {
    expect(err).toMatchSnapshot();
  }
});
