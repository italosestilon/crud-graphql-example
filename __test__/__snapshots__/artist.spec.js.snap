// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should delete an artist 1`] = `
Object {
  "__v": 0,
  "_id": "ObjectId",
  "albums": Array [],
  "name": "Lana Del Rey",
}
`;

exports[`should not save an empty artist 1`] = `[ValidationError: Artist validation failed: name: Path \`name\` is required.]`;

exports[`should retrieve an artist 1`] = `
Object {
  "__v": 0,
  "_id": "ObjectId",
  "albums": Array [],
  "name": "Lana Del Rey",
}
`;

exports[`should save new artist 1`] = `
Object {
  "__v": 0,
  "_id": "ObjectId",
  "albums": Array [],
  "name": "Lana Del Rey",
}
`;
